### ISSUE TITLE
Link disappears when filter applied

### ISSUE THREAD

**Issue by DorothyGuan** [clientIO/joint#1314]
* jointjs version: 3.1.1
* Platform: macOS Mojave Google Chrome 83.0.4103.61
* Description: 

I want to add some filter effects when I hover a link. So I add 'link:mouseenter' and 'link:mouseleave' listeners on the graph to change corresponding attrs. (I don't know whether this usage is recommended or not). 

Links work well if they are not straight. But if I move a node to make the link horizontal or vertical, it will disappear after hover action.

Here is a example:
[jointjs-demo.zip](https://github.com/clientIO/joint/files/4721938/jointjs-demo.zip)
(I'm sorry I can't provide an online reproduce link, so I wrote a html example to explain this question. )

**Comment by kumilingus** [comment]
Please see https://resources.jointjs.com/tutorial/links-patterns spec. **Pure vertical and horizontal lines**.

**Comment by DorothyGuan** [comment]
I read this section and define a custom connector which extends the built-in 'rounded' connector for this situation, then I find a new problem. The line won't disappear finally, but it gets slimmer after hover action. And the shadow seems not as obvious as a winding line acts.
When I adjust the 'width' or 'height', the shadow changes along with the 'slope' change. More sloping, more obvious the shadow is. Like these:

![屏幕快照 2020-06-03 下午6 54 23](https://user-images.githubusercontent.com/46178352/83629786-50c22180-a5cd-11ea-9835-880efaf66e22.png)
![屏幕快照 2020-06-03 下午7 05 56](https://user-images.githubusercontent.com/46178352/83629791-528be500-a5cd-11ea-83b6-bb90f3100817.png)
[jointjs-demo.html.zip](https://github.com/clientIO/joint/files/4722827/jointjs-demo.html.zip)


It seems not that perfect in a display scene. So Maybe I need to consider another way to solve this problem. However, Thanks for your reply 😊

**Comment by DorothyGuan** [comment]
Not find a better way, so I give up shadow filter and use stroke in replace.

**Comment by kumilingus** [comment]
Hi, sorry for late reply.
The problem is that we use `objectBoundingBox` for `filterUnits` [here](https://github.com/clientIO/joint/blob/master/src/dia/Paper.mjs#L2552-L2562). That means the size of the shadow depends on size of the node it applies to.

From **MDN:**
> objectBoundingBox
> In that case, x, y, width and height represent fractions or percentages of the bounding box on the referencing element.

I am marking this as a bug.

**Comment by cgiuliani** [comment]
hi,

I have exactly the same problem with horizontal links.
I use filters to make a double line link, but this bug happens.
Is there a solution or alternative to make my double line?

![2022-11-15_11-43](https://user-images.githubusercontent.com/24897654/201900676-e4a8bb32-7553-45d3-b908-76492bd315c1.png)
![2022-11-15_11-42](https://user-images.githubusercontent.com/24897654/201900689-fd85d2f5-4a42-418f-abd7-2b2669491fe9.png)


Thanks

**Comment by kumilingus** [comment]
> Please see https://resources.jointjs.com/tutorial/links-patterns spec. **Pure vertical and horizontal lines**.

This is a workaround.

And [joint.shapes.standard.DoubleLink](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#shapes.standard.DoubleLink) is an alternative.

**Comment by kumilingus** [comment]
Update: this seems to work nicely:
```js
link.attr('line/filter': {
    name: 'dropShadow',
    args: { dx: 2, dy: 2, blur: 3 },
    attrs: {
        filterUnits: 'userSpaceOnUse',
        // Reset to the default values
        x: '-10%',
        y: '-10%',
        width: '120%',
        height: '120%'
    }
});
```

**Comment by kumilingus** [comment]
Fixed in `v4.0.0`.
